clusterloader2/api/default.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"fmt"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+// SetDefaults set the default configuration parameters.
+func (conf *Config) SetDefaults() {
+	//To support deprecated "automanangedNamespaces" in testconfig
+	conf.AutomanagedNamespaces = 1
+
+	conf.Namespace.SetDefaults()
+}
+
+// SetDefaults specifies the default values for namespace parameters.
+func (ns *NamespaceConfig) SetDefaults() {
+	ns.Number = 1
```

Comment:
That is incorrect. You should default to 1, only if it was previously set to 0.

Reply:
@wojtek-t  I'm not sure about this one. My implementation is to set the defaults before parsing the testconfig yaml file. The default value will be overridden if it's set in testconfig.
I think it would be a neater implementation. If we set the defaults after unmarshalling the testconfig, we have to check if "0" or false is set or unset using pointers.  The same to the delete namespace flags. I already tested it with different cases and worked as expected. 

https://github.com/yuanchen8911/perf-tests/blob/master/clusterloader2/pkg/test/test.go
```
# Directory Structure

- profile: clusterloader template files for different test cases
- profile/override: test case-specific parameter files
- script: scirpts to run tests and manage kubernetes resources
- config: kubelets config files
```

Reply:
I comment on that too - I'm not a fan of this approach. I much more prefer the more standard approach (and what k8s itself is doing internally), which is what I suggested:
- read the config
- for fields that are unset, propagate with defaults

Reply:
My current implementation is to set the defaults first and override them if set later.

Reply:
OK, I'll make the changes.